akka-persistence-testkit/src/main/scala/akka/persistence/testkit/InMemStorage.scala

```diff
@@ -0,0 +1,417 @@
+        })
+
+    reduced match {
+      case ProcessingSuccess =>
+        add(elems)
+        Success(())
+      case Reject(ex)         => Failure(ex)
+      case StorageFailure(ex) => throw ex
+    }
+  }
+
+  def tryRead(
+      persistenceId: String,
+      fromSequenceNr: Long,
+      toSequenceNr: Long,
+      max: Long): immutable.Seq[PersistentRepr] = {
+    val batch = read(persistenceId, fromSequenceNr, toSequenceNr, max)
+    currentPolicy.tryProcess(persistenceId, ReadEvents(batch)) match {
+      case ProcessingSuccess  => batch
+      case Reject(ex)         => throw ex
+      case StorageFailure(ex) => throw ex
+    }
+  }
+
+  def tryReadSeqNumber(persistenceId: String): Long = {
+    currentPolicy.tryProcess(persistenceId, ReadSeqNum) match {
+      case ProcessingSuccess  => getHighestSeqNumber(persistenceId)
+      case Reject(ex)         => throw ex
+      case StorageFailure(ex) => throw ex
+    }
+  }
+
+  def tryDelete(persistenceId: String, toSeqNumber: Long): Unit = {
+    currentPolicy.tryProcess(persistenceId, DeleteEvents(toSeqNumber)) match {
+      case ProcessingSuccess  => deleteToSeqNumber(persistenceId, toSeqNumber)
+      case Reject(ex)         => throw ex
+      case StorageFailure(ex) => throw ex
+    }
+  }
+
+  private def mapAny(key: String, elems: immutable.Seq[Any]): immutable.Seq[PersistentRepr] = {
+    val sn = getHighestSeqNumber(key) + 1
+    elems.zipWithIndex.map(p => PersistentRepr(p._1, p._2 + sn, key))
+  }
+
+}
+
+object EventStorage {
+
+  object JournalPolicies extends DefaultPolicies[JournalOperation]
+
+}
+
+/**
+ * INTERNAL API
+ *
+ * Persistent journal operations.
+ */
+@InternalApi
+sealed trait JournalOperation
+
+/**
+ * Read from journal operation with events that were read.
+ */
+final case class ReadEvents(batch: immutable.Seq[Any]) extends JournalOperation {
+
+  def getBatch(): JList[Any] = batch.asJava
+
+}
+
+/**
+ * Write in journal operation with events to be written.
+ */
+final case class WriteEvents(batch: immutable.Seq[Any]) extends JournalOperation {
+
+  def getBatch(): JList[Any] = batch.asJava
+
+}
+
+/**
+ * Read persistent actor's sequence number operation.
+ */
+case object ReadSeqNum extends JournalOperation {
+
+  /**
+   * Java API: the singleton instance.
+   */
+  def getInstance() = this
+
+}
+
+/**
+ * Delete events in the journal up to `toSeqNumber` operation.
+ */
+final case class DeleteEvents(toSeqNumber: Long) extends JournalOperation {
+
+  def getToSeqNumber() = toSeqNumber
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[testkit] class SimpleEventStorageImpl extends EventStorage {
+
+  override type InternalRepr = PersistentRepr
+
+  override def toInternal(repr: PersistentRepr): PersistentRepr = repr
+
+  override def toRepr(internal: PersistentRepr): PersistentRepr = internal
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[testkit] class SerializedEventStorageImpl(system: ActorSystem) extends EventStorage {
```

Comment:
This file is relatively large, perhaps move a couple of these implementation classes to their own files under an `internal` package?

Reply:
made more splits and moves